subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -689,4 +681,131 @@ static struct lll_df_adv_cfg *df_adv_cfg_acquire(void)
+	pdu_data.extra_data = df_cfg;
+
+	/* Note: ULL_ADV_PDU_HDR_FIELD_CTE_INFO is just information that extra_data
+	 * is required in case of this ull_adv_sync_pdu_alloc.
+	 * Other flags here do not change anything. It may be changed to use
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0, &pdu_data, &pdu_prev,
+				     &pdu, &extra_data_prev, &extra_data, ter_idx);
+	if (err) {
+		return err;
+	}
+
+	if (extra_data) {
+		ull_adv_sync_extra_data_set_clear(extra_data_prev,
+						  extra_data,
+						  ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+						  0, &pdu_data);
+	}
+
+	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+#if defined(CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX) && (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1)
+					 (df_cfg->cte_count > 1) ?
+						       ULL_ADV_PDU_HDR_FIELD_CTE_INFO |
+							 ULL_ADV_PDU_HDR_FIELD_AUX_PTR :
+#endif /* CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX && CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 */
+						       ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+					 0, &pdu_data);
+	if (err) {
+		return err;
+	}
+
+#if defined(CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX) && (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1)
+	uint8_t pdu_add_field_flags;
+	struct pdu_adv *pdu_next;
+	uint8_t cte_index = 1;
+	bool new_chain;
+
+	new_chain = (pdu_prev == pdu ? false : true);
+
+	pdu_add_field_flags = ULL_ADV_PDU_HDR_FIELD_CTE_INFO;
+
+	pdu_prev = lll_adv_pdu_linked_next_get(pdu_prev);
+
+	/* Update PDUs in existing chain. Add cte_info to extended advertising header. */
+	while (pdu_prev != NULL) {
+		if (new_chain == true) {
+			pdu_next = lll_adv_pdu_alloc_pdu_adv();
+			lll_adv_pdu_linked_append(pdu_next, pdu);
+			pdu = pdu_next;
+		} else {
+			pdu = lll_adv_pdu_linked_next_get(pdu);
+		}
+
+		pdu_next = lll_adv_pdu_linked_next_get(pdu_prev);
+		/* If all CTEs were added to chain, remove CTE from flags */
+		if (cte_index >= df_cfg->cte_count) {
+			pdu_add_field_flags = 0;
+		} else {
+			++cte_index;
+			if (!pdu_next && cte_index < df_cfg->cte_count) {
+				pdu_add_field_flags |= ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
+			}
+		}
+
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu, pdu_add_field_flags, 0,
+						 &pdu_data);
+		if (err != 0) {
+			/* TODO: implement gracefull error handling, cleanup of
+			 * changed PDUs and notify host abut issue during start of CTE
+			 * transmission.
+			 */
+			return err;
+		}
+		pdu_prev = pdu_next;
+	}
+
+	/* If there is missing only one CTE do not add aux_ptr to PDU */
+	if (df_cfg->cte_count - cte_index >= 2) {
+		pdu_add_field_flags |= ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
+	} else {
+		pdu_add_field_flags = ULL_ADV_PDU_HDR_FIELD_CTE_INFO;
+	}
+
+	/* Add new PDUs if the number of PDUs in existing chain is lower than requested number
+	 * of CTEs.
+	 */
+	while (cte_index < df_cfg->cte_count) {
+		pdu_prev = pdu;
+		pdu = lll_adv_pdu_alloc_pdu_adv();
+		if (!pdu) {
+			return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+		}
```

Comment:
I guess this also needs a todo similar to the TODO in lines 768-770

Reply:
Yes, there is missing TODO about graceful error handling. The host is notified about problem here by return of `BT_HCI_ERR_MEM_CAPACITY_EXCEEDED`